### PR TITLE
Add basic GCN Events page

### PR DESCRIPTION
**Pull request by acrellin**
A rough draft to be expanded upon by @mcoughlin and Solene.

gcn_events_table

gcn_sidebar

### CODE REVIEW

**Comment by mcoughlin** [comment]
@acrellin we could also generalize the Tags to use that color scheme from #2082, but this is awesome.

**Comment by acrellin** [comment]
Yeah, I was aiming for as simple as possible here as a base for you to
build on and tweak.

**Comment by mcoughlin** [comment]
Yes to be clear, this was just me thinking this through. I am happy with this!

**Review by mcoughlin — APPROVED** [review]

**Review by kmshin1397 — COMMENTED** [review]
Regarding the merge conflict here @acrellin , I renamed the redux action `FETCH_RECENT_GCN_EVENTS` in the widget PR because that's really what the handler logic does. But with this PR, I think it may be time to do some refactoring to move that logic to a new `/api/gcn_event/recent` and implement an actual general `GET multiple /api/gcn_event` handler to be used by the data table which could be extended in the future to support filtering, pagination, etc. What do you think?

**Comment by acrellin** [comment]
@kmshin1397 I didn't see the changes you made to that handler after I approved it. I think that should've been changed to a different endpoint/handler name when it was changed from fetching all events to just the most recent. Perhaps an intermediate PR is called for to get master in better shape in those respects. Thoughts?

**Comment by acrellin** [comment]
I can do that here, too, I suppose, I just think what went in in that PR is not quite right.

**Comment by kmshin1397** [comment]
Yeah I agree, I didn't really think the other PR through enough. The intermediate PR is probably better - I can work on that right now.

**Comment by acrellin** [comment]
Thanks a lot @kmshin1397 !

**Comment by acrellin** [comment]
@kmshin1397 I moved the recent events stuff to its own redux module

File: static/js/ducks/gcnEvents.js [hunk 1]
```diff
@@ -3,22 +3,23 @@ import messageHandler from "baselayer/MessageHandler";
...
-export const FETCH_RECENT_GCNEVENTS = "skyportal/FETCH_RECENT_GCNEVENTS";
-export const FETCH_RECENT_GCNEVENTS_OK = "skyportal/FETCH_RECENT_GCNEVENTS_OK";
+const FETCH_GCN_EVENTS = "skyportal/FETCH_GCN_EVENTS";
+const FETCH_GCN_EVENTS_OK = "skyportal/FETCH_GCN_EVENTS_OK";
 
-export const fetchRecentGcnEvents = () =>
-  API.GET("/api/internal/recent_gcn_events", FETCH_RECENT_GCNEVENTS);
+// eslint-disable-next-line import/prefer-default-export
+export const fetchGcnEvents = () =>
+  API.GET("/api/gcn/event_views", FETCH_GCN_EVENTS);
```
Comment (kmshin1397) [inline static/js/ducks/gcnEvents.js:11]: I think this API route might be out-of-date? It's api/gcn_events right?

**Review by kmshin1397 — COMMENTED** [review]
I have the one comment, but otherwise looks good!

File: static/js/ducks/gcnEvents.js (hunk 1 above)
Comment (acrellin) [reply static/js/ducks/gcnEvents.js]: yeah, thanks. updated

**Comment by acrellin** [comment]
When https://github.com/skyportal/skyportal/pull/2090 is merged, this component will be updated such that the individual rows (events) link to the event details page

**Comment by mcoughlin** [comment]
@acrellin Now that #2090 is merged, maybe this one can be updated and get this one in as you have time.

**Comment by acrellin** [comment]
@mcoughlin @kmshin1397 This PR has been updated such that the obs date in the table now links to the individual event page: 
gcn_events_links

gcn_event

**Comment by mcoughlin** [comment]
Thanks @acrellin !!

**Review by kmshin1397 — APPROVED** [review]

**Pull request merged**